auth/standard_role_manager.cc

```diff
@@ -267,6 +267,7 @@ future<> standard_role_manager::start() {
                 }
                 co_await create_default_role_if_missing();
             });
+            return _stopped.then([] {});
```

Comment:
I'm not sure what `.then([] {})` is supposed to do?

Reply:
Well. After attaching a `then` to a future, the future becomes unusable. (Waiting for it again or trying to attach a different continuation is a bug, and IIRC it would trigger some assert in seastar.)

So it is basically equivalent to `return std::move(_stopped)`. But using `std::move` makes it a bit clearer that this future is no longer supposed to be used.

Reply:
Probably even better: `return std::exchange(_stopped, make_ready_future<>())`